### PR TITLE
Added tutorial page for Visualization and Analysis

### PR DESCRIPTION
**Pull request by zhuokaizhao**
Thank you for reviewing.

### CODE REVIEW

**Comment by cryos** [comment]
Looks good, please don't worry about adding the table of contents manually - it is creates automatically, see:
image
Are you previewing this locally? For me none of the images show up, and I would prefer you to stick with markdown for images, e.g.
```
![Open data file](img/tomviz_open_data.png) 
<img src="img/tomviz_open_data.png" height="250">
```
Top one not the bottom one.

**Comment by zhuokaizhao** [comment]
Manually generated table of contents has been removed, image problems have been fixed. Thanks.

File: docs/viz.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+Typical datasets may have following relationships between attributes and sizes.
+
+| Volume          | Voxels        | Size (char) | Size (int) |
+|  :---           |  :---         |    :---     | :---       |
+| 64<sup>3</sup>  | 262,144       | 0.25 MB     | 1 MB       |
+| 128<sup>3</sup> | 2,097,152     | 2.00 MB     | 8 MB       |
+| 256<sup>3</sup> | 16,777,216    | 16.00 MB    | 64 MB      |
+| 512<sup>3</sup> | 134,217,728   | 128.00 MB   | 512 MB     |
+| 103<sup>3</sup> | 1,073,741,824 | 1,024.00 MB | 4096 MB    |
```
Comment (cryos) [inline docs/viz.md]: This should be 1024 (reads 103)

File: docs/viz.md [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+## Visualization techniques
+
+In this section, we will go over some available techniques and explain the important parameters. Most of the techniques are GPU accelerated, which requires a good graphics card that has at least 1GB memory.
+
+### Available visualization modules
+
+![viz_modules](img/viz_modules.png)
+
+Visualization modules were implemented and optimized using C++ and GLSL (GLslang) to take full advantage of hardware accelerations. These modules are available from the visulization menu.
```
Comment (cryos) [inline docs/viz.md]: Not sure what `GLslang` in brackets is, suggest removing it. We can link to GLSL in a future iteration if people want to know more about. There are typos such as `visulization` that I don't think were in the original. If using VS Code you can install a spell checker plugin and run it over this file.

File: docs/viz.md [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+
+![raw_reader](img/raw_reader.png)
+
+### Save results
+
+#### Save data
+
+Users can save the data by either clicking the ```Save Data``` button from ```File``` (as shown below), or simply using the keyboard short-cut ```Ctrl+S```.
+
+![tomviz_save_data](img/tomviz_save_data.png)
```
Comment (cryos) [inline docs/viz.md]: In general it is a shame you didn't carry over the alt text from the slides, these all appear to be the file name with out the suffix. We can add these back in and maybe improve them in a future iteration.

File: docs/viz.md [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+![default_view](img/default_view.png)
+
+#### Palette and background colors
+
+Palette can be easily modified with several presets.
+
+![palette](img/palette.png)
+
+Tips: balck blackground is recommended in some situations on monitors; white is preferred for print, web pages and etc. to highlight samples.
```
Comment (cryos) [inline docs/viz.md]: Typo - balck should be black

File: docs/viz.md [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+![volume_2d_transfer](img/volume_2d_transfer.png)
+
+Besides it, 2D transfer function provides more selectivity for some data sets. As shown below, users can select a box in the 2D histogram.
+
+![volume_2d_transfer2](img/volume_2d_transfer2.png)
+
+<!--*********************** Section 4 Data Transforms ***********************-->
+## Data Transforms
+
+Data transforms were implemented as operators in C++ or Python. They can be applied to the data directlry, while some of them provide graphical interfaces. The operators source code can be viewed and edited from the ```Context``` menu. Most of the operators make extensive use of NumPy.
```
Comment (cryos) [inline docs/viz.md]: directlry should be directly

**Comment by cryos** [comment]
There are some typos I pointed out inline (may be more, these are just the ones I noticed). This looks much better in the generated pages. Once finished it would be good to move on to the other sections following this pattern - with them please try to keep the alt text in the slide source too.

File: docs/viz.md (hunk 2 above)
Comment (zhuokaizhao) [reply docs/viz.md]: I was not aware of GLSL, so I searched for it and it said "GLSL (GLslang) is a short term for the official OpenGL Shading Language.", which is why I put GLslang in the brackets. I can remove it though, if the abbreviation itself is better self-explained. 

Thank you for letting me know that VS Code has spell checker plugin, I just installed it and fixed all the typos.

File: docs/viz.md (hunk 3 above)
Comment (zhuokaizhao) [reply docs/viz.md]: Oh I did not know those names within the brackets were there for a reason, I will copy them over then. Thanks.

Comment (zhuokaizhao) [reply docs/viz.md]: Just to make sure, the `alt text` you mean the text inside the brackets right?

File: docs/viz.md (hunk 2 above)
Comment (cryos) [reply docs/viz.md]: I think the abbreviation can just link to a page on GLSL definition, it is not central to the text other than to point out that things are hardware accelerated.

File: docs/viz.md (hunk 3 above)
Comment (cryos) [reply docs/viz.md]: Yes - `Save data` where `[Save data]` is the image alt text, and `(/images/tomviz_save_data.png)` is the path to the image.

**Pull request merged**